### DISCUSSION TITLE
Query function to get value of a label

### DISCUSSION THREAD

**Discussion by urusha** [prometheus/prometheus#10620] [Ideas]
## Proposal
Query function to get value of a label. This would be convenient to have alert thresholds stored in labels. What would lead to simplified alert rules.
Example:
Metrics:
```
metric1{label1="foo",label2="10"}
metric1{label1="bar",label2="20"}
```
Alert rule:
`metric1 < label_value(metric1,label2)`
would turn into:
```
metric1{label1="foo",label2="10"} < 10
metric1{label1="bar",label2="20"} < 20
```

**Comment by prymitive** [comment]
A common pattern is to export a time series to act as threshold - https://www.robustperception.io/using-time-series-as-alert-thresholds

**Reply by roidelapluie** [reply]
I second this.
